### PR TITLE
fixes dotnet/templating#2746 when using source modifiers returned file changes are wrong

### PR DESCRIPTION
**Pull request by vlada-shubina**
fixes dotnet/templating#2746 when using source modifiers returned file changes are wrong
reenabled disabled tests

### CODE REVIEW

**Comment by vlada-shubina** [comment]
@dotnet/templating-engine-maintainers please check only last commit, the rest is covered in https://github.com/dotnet/templating/pull/2690.

Two points to consider:
- this PR modifies `IOrchestator` interfaces, probably we need to create IOrchestrator3 one, however I cannot fix the issue for 1 and 2 then
- alternative is changing `IGlobalRunSpec` interface

**Review by DavidKarlas — APPROVED** [review]
Changes seem, sane, also nice unit tests... Just need to check if it breaks anything out there...

**Comment by DavidKarlas** [comment]
I think its fine changing this interface... And I would even go step further and drop `2` since its only used by RunnableProjectGenerator...

File: src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/FileRenameGenerator.cs [hunk 1]
```diff
@@ -46,6 +46,15 @@ internal static class FileRenameGenerator
             return allRenames;
         }
 
+        public static string ApplyRenameToPrimaryOutput(string primaryOutputPath, IEngineEnvironmentSettings environmentSettings, string sourceName, object resolvedNameParamValue, IParameterSet parameterSet, IReadOnlyList<IReplacementTokens> symbolBasedFileRenames = null)
+        {
+            string targetDirectoryStub = string.Empty;
+            IProcessor symbolRenameProcessor = SetupSymbolBasedRenameProcessor(environmentSettings, sourceName, ref targetDirectoryStub, resolvedNameParamValue, parameterSet, symbolBasedFileRenames);
+            string renameFinalTargetValue = ApplyRenameProcessorToFilename(symbolRenameProcessor, primaryOutputPath);
+
+            return renameFinalTargetValue;
```
Comment (donJoseLuis) [inline src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/FileRenameGenerator.cs:55]: minor: Could IProcessor include this state so as to not need the param?

**Review by donJoseLuis — APPROVED** [review]

**Pull request merged**